rln/src/lib.rs

```diff
@@ -25,116 +20,108 @@ mod test {
     use hex_literal::hex;
     use num_bigint::BigInt;
     use semaphore::{
-        hash::Hash, hash_to_field, identity::Identity, poseidon_tree::PoseidonTree, Field,
+        hash::Hash, hash_to_field, identity::Identity, poseidon_hash, poseidon_tree::PoseidonTree,
+        Field,
     };
 
     #[test]
+    // We test Merkle Tree generation, proofs and verification
     fn test_merkle_proof() {
-        let leaf = Field::from(0);
+        let tree_height = 16;
+        let leaf_index = 3;
 
         // generate identity
-        let id = Identity::from_seed(b"hello");
+        // We follow zk-kit approach for identity generation
+        let id = Identity::from_seed(b"test-merkle-proof");
+        let identity_secret = poseidon_hash(&vec![id.trapdoor, id.nullifier]);
+        let id_commitment = poseidon_hash(&vec![identity_secret]);
 
         // generate merkle tree
-        let mut tree = PoseidonTree::new(21, leaf);
-        tree.set(0, id.commitment());
-
-        let merkle_proof = tree.proof(0).expect("proof should exist");
-        let root: Field = tree.root().into();
+        let default_leaf = Field::from(0);
+        let mut tree = PoseidonTree::new(tree_height, default_leaf);
+        tree.set(leaf_index, id_commitment.into());
 
-        println!("Root: {:#}", root);
-        println!("Merkle proof: {:#?}", merkle_proof);
+        // We check correct computation of the root
+        let root = tree.root();
+        assert_eq!(
+            root,
+            Field::from_str("0x27401a4559ce263630907ce3b77c570649e28ede22d2a7f5296839627a16e870")
```

Comment:
Is this from somewhere specifically?

Reply:
No, was randomly generated by zk-kit implementation.